File: UTESA/DCE 5 Estrellas/dce2005/Estrella1/DCE1_Ejemplos/src/VB/Cliente/OpcionListarCarrito.vb
Imports DCE05.Ejemplos.EstrellaUno.ReglasNegocio

''' <summary>
''' Representa la opción de listar el carrito de compras.
''' </summary>
Public Class OpcionListarCarrito
    Inherits Opcion

    ''' <summary>
    ''' Construye una instancia de la opción con sus datos básicos.
    ''' </summary>
    ''' <param name="codigo">El código de la opción.</param>
    Friend Sub New(ByVal Codigo As Integer)
        Me.Codigo = Codigo
        Descripcion = "Listar Carrito de Compras"
    End Sub

    ''' <summary>
    ''' Ejecuta la acción asociada a la opción.
    ''' </summary>
    ''' <exception cref="OpcionInvalidaException">Si la opción no fue ejecutada exitosamente.</exception>
    Friend Overrides Sub EjecutarAccion()
        If PuntoDeVenta.VentaActual Is Nothing Then
            Throw New OpcionInvalidaException("La venta no fue iniciada.")
        End If

        If PuntoDeVenta.VentaActual.Items.Count.Equals(0) Then
            Console.WriteLine("Carrito vacío.")
        Else
            Console.WriteLine("Carrito de Compras")
            Console.WriteLine("------------------" + Chr(13))

            Console.WriteLine("Items: " + PuntoDeVenta.VentaActual.Items.Count.ToString())
            Console.WriteLine("Codigo".PadRight(7) + Chr(7) + "Producto".PadRight(30) + Chr(7) + "Cantidad".PadRight(9) + Chr(7) + "Sub Total")

            For Each Item As ItemVenta In PuntoDeVenta.VentaActual.Items
                Console.WriteLine(Item.Codigo.ToString().PadRight(7) + Chr(7) + Item.Producto.Descripcion.PadRight(30) + Chr(7) + Item.Cantidad.ToString().PadRight(9) + Chr(7) + Item.CalcularTotal().ToString())
            Next
            Console.WriteLine(Chr(13))
        End If
    End Sub
End Class

File: UTESA/DCE 5 Estrellas/dce2005/Estrella1/DCE1_Ejemplos/src/CS/Cliente/OpcionListarCarrito.cs
using System;
using System.Collections.Generic;
using System.Text;

using DCE05.Ejemplos.EstrellaUno.ReglasNegocio;

namespace DCE05.Ejemplos.EstrellaUno.Cliente {

    /// <summary>
    /// Representa la opción de listar el carrito de compras.
    /// </summary>
    internal class OpcionListarCarrito : Opcion {

        /// <summary>
        /// Construye una instancia de la opción con sus datos básicos.
        /// </summary>
        /// <param name="codigo">El código de la opción.</param>
        internal OpcionListarCarrito(int codigo) {
            Codigo = codigo;
            Descripcion = "Listar Carrito de Compras";
        }

        /// <summary>
        /// Ejecuta la acción asociada a la opción.
        /// </summary>
        /// <exception cref="OpcionInvalidaException">Si la opción no fue ejecutada exitosamente.</exception>
        internal override void EjecutarAccion() {
            
            if (PuntoDeVenta.VentaActual == null) {
                throw new OpcionInvalidaException("La venta no fue iniciada.");
            }

            if (PuntoDeVenta.VentaActual.Items.Count.Equals(0)) {
                Console.WriteLine("Carrito vacío.");
            } else {
                Console.WriteLine("Carrito de Compras");
                Console.WriteLine("------------------\n");

                Console.WriteLine("Items: " + PuntoDeVenta.VentaActual.Items.Count.ToString());
                Console.WriteLine("{0}\t{1}\t{2}\t{3}",
                    "Codigo".PadRight(7),
                    "Producto".PadRight(30),
                    "Cantidad".PadRight(9),
                    "Sub Total");

                foreach (ItemVenta item in PuntoDeVenta.VentaActual.Items) {
                    Console.WriteLine("{0}\t{1}\t{2}\t{3}",
                        item.Codigo.ToString().PadRight(7),
                        item.Producto.Descripcion.PadRight(30),
                        item.Cantidad.ToString().PadRight(9),
                        item.CalcularTotal());
                }
            }
        }

    }
}
